lib/mosaic-utils/src/main/java/org/eclipse/mosaic/lib/util/gson/UnitFieldAdapter.java

```diff
@@ -167,5 +173,91 @@ private double determineUnitMultiplier(String prefix, String unit) {
             return (TypeAdapter<T>) new UnitFieldAdapter(false, SPEED_PATTERN);
         }
     }
+
+    public static class WeightKiloGrams implements TypeAdapterFactory {
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+            return (TypeAdapter<T>) new UnitFieldAdapter(true, WEIGHT_PATTERN) {
+                @Override
+                double determineUnitMultiplier(String prefix, String unit) {
+                    if (unit.startsWith("mile")) {
+                        //special case
+                        return 1609.344;
+                    } else if (StringUtils.isNotEmpty(prefix)) {
+                        return 0.001 * UNIT_MULTIPLIERS.getOrDefault(prefix, 1d);
+                    }
+                    return 1d;
```

Comment:
Should this return 0.001 instead? At least this is different to the behavior of `WeightKiloGramsQuiet`

Reply:
Yes! I was stuck on this for a little bit, but since kg is the SI-unit I think we want it the way it is in `WeightKiloGramsQuiet`